src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.cs

```diff
@@ -44,6 +44,13 @@ public static partial class PlatformDetection
 
         public static bool IsNotNetNative => !IsNetNative;
 
+        // Windows - Schannel supports alpn from win8.1/2012 R2 and higher.
+        // Linux - OpenSsl supports alpn from openssl 1.0.2 and higher.
+        // OSX - SecureTransport doesn't expose alpn APIs. #30492
+        public static bool BackendSupportsAlpn => (IsWindows && !IsWindows7) ||
```

Comment:
Will this property be false on OSX since #30492 prevents ALPN right now on OSX? I only see the comparison about Windows OS versions and Linux and OpenSsl versions.

Reply:
If it's OSX, it's like `(false) || (false)`, so the property should be evaluated as false.

My PR here https://github.com/dotnet/corefx/pull/31440 has no OSX failure, is a testing for the property.